docs/4.4/api-reference.md

```diff
@@ -11,57 +11,326 @@ Teleport is currently working on documenting our API.
+    Rules: []services.Rule{
+      // - resources: ['session']
+      //   verbs: ['list', 'read']
+      services.NewRule(services.KindSession, services.RO()),
+    },
+  },
+  Deny: services.RoleConditions{
+    // node_labels: '*': '*'
+    NodeLabels: services.Labels{"*": []string{"*"}},
+  },
+})
+if err != nil {
+  return err
+}
+
+if err = client.UpsertRole(ctx, role); err != nil {
+  return err
+}
+```
+
+**Update role**
+
+```go
+// update the auditor role's ttl to one day
+role.SetOptions(services.RoleOptions{
+	MaxSessionTTL: services.Duration(time.Hour * 24),
+})
+if err := client.UpsertRole(ctx, role); err != nil {
+  return err
+}
+```
 
-## Audit Events API
-Coming Soon
+**Delete role**
+
+```go
+if err := client.DeleteRole(ctx, "auditor"); err != nil {
+  return err
+}
+```
+
+## Tokens
+
+[Tokens](admin-guide.md#adding-nodes-to-the-cluster) are used to add nodes to clusters, and add remote clusters to [trusted clusters](admin-guide.md#trusted-clusters).
+
+**Retrieve token**
+
+```go
+token, err := client.GetToken(tokenString)
+if err != nil {
+  return err
+}
+```
+
+**Create token**
+
+```go
+tokenString, err := client.GenerateToken(ctx, auth.GenerateTokenRequest{
+  // Token: "custom token value", // Set token explicitly, otherwise it will be generated
+  Roles: teleport.Roles{teleport.RoleProxy},
+  TTL:   time.Hour,
+})
+if err != nil {
+  return err
+}
+```
+
+**Update token**
+
+```go
+// updates the token to be a proxy token
+token.SetRoles(teleport.Roles{teleport.RoleProxy})
+if err := client.UpsertToken(token); err != nil {
+  return err
+}
+```
+
+**Delete token**
+
+```go
+if err := client.DeleteToken(tokenString); err != nil {
+  return err
+}
+```
+
+## Cluster Labels
+
+The root cluster in a [trusted cluster](admin-guide.md#adding-nodes-to-the-cluster) can add/update cluster labels on remote clusters.
+
+**Create a remote cluster token with Labels**
+
+Remote clusters will inherit the labels of the [join token](trustedclusters.md#join-tokens) used to add them to the trusted cluster. This is the preferred method of managing cluster tokens.
+
+Follow the trusted cluster join token [docs](trustedclusters.md#join-tokens) to use this token to create a remote cluster using `tctl`.
+
+```go
+tokenString, err := client.GenerateToken(ctx, auth.GenerateTokenRequest{
+  // You can provide 'Token' for a static token name
+  Roles: teleport.Roles{teleport.RoleTrustedCluster},
+  TTL:   time.Hour,
+  Labels: map[string]string{
+    "env": "staging",
+  },
+})
+```
+
+**Update remote cluster's labels**
+
+You can also update a remote cluster's labels from the root cluster using the auth api if necessary.
+
+```go
+rc, err := client.GetRemoteCluster("remoteClusterName")
+if err != nil {
+  return err
+}
+
+md := rc.GetMetadata()
+md.Labels = map[string]string{"env": "prod"}
+rc.SetMetadata(md)
+
+if err = client.UpdateRemoteCluster(ctx, rc); err != nil {
+  return err
+}
+
+```
+
+## Access Requests
```

Comment:
Access Workflows API

Reply:
This has reminded me to update it in the side nav, I'll do this when I help @Joerger deploy this.